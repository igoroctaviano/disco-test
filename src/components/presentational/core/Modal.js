import React from "react";
import PropTypes from 'prop-types';

const Modal = ({ isActive, onClose, itemRenderer }) =>
  isActive ? (
    <div className="modal">
      <div className="modal-content">
        <span onClick={onClose} className="close-button"></span>
        {itemRenderer()}
      </div>
    </div>
  ) : null;

Modal.propTypes = {
  isActive: PropTypes.bool,
  onClose: PropTypes.func,
  itemRenderer: PropTypes.func
};

export default Modal;
