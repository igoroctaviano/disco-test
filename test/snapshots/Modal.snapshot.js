import React from "react";
import Modal from "./../../src/components/presentational/core/Modal.js";
import renderer from "react-test-renderer";
import { items, itemRenderer } from "./mocks.js";

test("renders correctly", () => {
  const tree = renderer
    .create(<Modal isActive itemRenderer={itemRenderer} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("doesnt render when not active", () => {
  const tree = renderer.create(<Modal itemRenderer={itemRenderer} />).toJSON();
  expect(tree).toEqual(null);
});
