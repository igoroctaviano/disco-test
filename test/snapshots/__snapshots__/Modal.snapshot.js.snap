// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal-content"
  >
    <span
      className="close-button"
    />
  </div>
</div>
`;
